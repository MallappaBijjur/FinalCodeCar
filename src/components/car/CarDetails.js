import React, { useState, useEffect } from 'react';
import '../home-car.css';
import TopHeader from '../common/TopHeader';
import Footer from '../common/Footer';
import Button from 'react-bootstrap/Button';
import { getOneCarInfo } from '../../helper/Service';

const CarDetails = ({ stockNumber }) => {
	const [ carDetails, setCarDetails ] = useState({});

	useEffect(() => {
		getOneCarInfo(stockNumber).then( ({ data }) => {
			const { car } = data;
			setCarDetails(car);
		});
	}, [stockNumber]);

	const addFavorite = carInfo => console.log(carInfo);

	return(
		<main className="car__details">
	        <TopHeader />
	        <section className="car__details--content">
	        	<div className='car__details--img' >
	                 <img
	                  src={carDetails && carDetails.pictureUrl} alt={carDetails && carDetails.modelName}
	                />
	             </div>
	             <div className='car-view'>
		          <div className='details'>
		            <h1
		              className='details__text--xlarge details__text--margin'
		            >
		              {carDetails && carDetails.modelName}
		            </h1>
		            <h2
		              className='details__text--large details__text--margin'
		            >
		              {`Stock #
		              ${carDetails && carDetails.stockNumber} -
		              ${carDetails && carDetails.mileage ? `${carDetails && carDetails.mileage.number} ${carDetails && carDetails.mileage.unit.toUpperCase()}` : 'none'}-
		              ${carDetails && carDetails.fuelType} -
		              ${carDetails && carDetails.color}` }
		            </h2>
		            <p className='details__text--small details__text--margin'>
		            This car is currently available and can be delivered as soon as
		            tomorrow morning. Please be aware that delivery times shown in
		            this page are not definitive and may change due to bad weather
		            conditions.
		            </p>
		          </div>
		          <div className='save-car'>
		            <div className='save-car__text'>
		              <p className='save-car__text--small'>
		              If you like this car, click the button and
		              save it in your collection of favourite
		              items.
		              </p>
		            </div>
		            <div className='save-car__btn--wrapper'>	              
		              <Button className="filter-button" onClick={() => addFavorite(carDetails)}>Save</Button>
		            </div>
		          </div>
	        </div>
	        </section>
	        <Footer />
	      </main>);
}

export default CarDetails;