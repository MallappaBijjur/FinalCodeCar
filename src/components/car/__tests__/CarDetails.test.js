import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, waitForElement } from '@testing-library/react';
import rerender from 'react-test-renderer';
import CarDetails from '../CarDetails';

let mockPromiseGetCarInfo;

jest.mock('../../../helper/Service', ()=> ({
    getOneCarInfo: jest.fn().mockImplementation(() => mockPromiseGetCarInfo)
}));

describe("CarDetails", () => {

	const originalError = console.error
	beforeAll(() => {
		console.error = (...args) => {
			if (/Warning.*not wrapped in act/.test(args[0])) {
				return
			}
			originalError.call(console, ...args)
		}
	})

	afterAll(() => {
		console.error = originalError
	});

	afterEach(cleanup);
	
    const carsData = {
		"car": {
					"stockNumber":30562,
					"manufacturerName":"Fiat",
					"modelName":"Ducato",
					"color":"yellow",
					"mileage":{
						 "number":100013,
						 "unit":"km"
					},
					"fuelType":"Petrol",
					"pictureUrl":"https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
			 }
	};
    
	it("renders without crashing", ()=>{
        mockPromiseGetCarInfo = Promise.resolve({ data: carsData });
		const div = document.createElement("div");
		ReactDOM.render(<CarDetails  stockNumber={30562} />, div);
	});
	
	it('renders Cars Details and Ducato as Model Name', async () => {
        mockPromiseGetCarInfo = Promise.resolve({ data: carsData });
        const { getByText } = render(<CarDetails stockNumber={30562} />);   
		const linkElement = await waitForElement(() => getByText("Ducato"));
		expect(linkElement).toBeInTheDocument();
	});
	
	it("matches snapshot", ()=> {
		const tree = rerender.create(<CarDetails stockNumber={30562} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});




