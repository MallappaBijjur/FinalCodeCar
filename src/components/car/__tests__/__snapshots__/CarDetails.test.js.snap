// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarDetails matches snapshot 1`] = `
<main
  className="car__details"
>
  <header
    className="header"
  >
    <div>
      <img
        alt="Auto1LOgo"
        className="header__logo"
        src="auto1logo.jpg"
      />
    </div>
    <div
      className="header__nav"
      data-testid="header"
    >
      <a
        className="header__navlink"
        href="#purchase"
      >
        Purchase
      </a>
      <a
        className="header__navlink"
        href="#myOrders"
      >
        My Orders
      </a>
      <a
        className="header__navlink"
        href="#Sell"
      >
        Sell
      </a>
    </div>
  </header>
  <section
    className="car__details--content"
  >
    <div
      className="car__details--img"
    >
      <img />
    </div>
    <div
      className="car-view"
    >
      <div
        className="details"
      >
        <h1
          className="details__text--xlarge details__text--margin"
        />
        <h2
          className="details__text--large details__text--margin"
        >
          Stock #
		              undefined -
		              none-
		              undefined -
		              undefined
        </h2>
        <p
          className="details__text--small details__text--margin"
        >
          This car is currently available and can be delivered as soon as tomorrow morning. Please be aware that delivery times shown in this page are not definitive and may change due to bad weather conditions.
        </p>
      </div>
      <div
        className="save-car"
      >
        <div
          className="save-car__text"
        >
          <p
            className="save-car__text--small"
          >
            If you like this car, click the button and save it in your collection of favourite items.
          </p>
        </div>
        <div
          className="save-car__btn--wrapper"
        >
          <button
            className="filter-button btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </section>
  <footer
    className="footer"
  >
    <p
      className="footer__text"
    >
      Â© AUTO1 Group 2018
    </p>
  </footer>
</main>
`;
